uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/sickness-leave-request-instance_test.js

```diff
@@ -0,0 +1,163 @@
+    describe('init', function () {
+      it('sickness request', function () {
+        expect(instance).toBeDefined();
+      });
+    });
+
+    describe('create()', function () {
+      beforeEach(function () {
+        requestData = helper.createRandomSicknessRequest();
+        instance = instance.init(requestData, false);
+        promise = instance.create();
+      });
+
+      afterEach(function () {
+        //to excute the promise force an digest
+        $rootScope.$apply();
+      });
+
+      it('calls equivalent API method', function () {
+        promise.then(function () {
+          expect(LeaveRequestAPI.create).toHaveBeenCalled();
+        });
+      });
+
+      it('id is appended to instance', function () {
+        expect(instance.id).not.toBeDefined();
+        promise.then(function () {
+          expect(instance.id).toBeDefined();
+          expect(instance.id).toEqual(jasmine.any(String));
+        });
+      });
+
+      describe('when one mandatory field is missing', function () {
+        beforeEach(function () {
+          expectedError = 'contact_id, from_date and from_date_type in params are mandatory';
+          delete instance.contact_id;
+          promise = instance.create();
+        });
+
+        afterEach(function () {
+          //to excute the promise force an digest
+          $rootScope.$apply();
+        });
+
+        it('fails to create instance', function () {
+          promise.catch(function (error) {
+            expect(error).toBe(expectedError);
+          });
+        });
+      });
+    });
+
+    describe('isValid()', function () {
+      beforeEach(function () {
+        requestData = {
+          contact_id: '123'
+        };
+        instance = instance.init(requestData);
+        promise = instance.isValid();
+      });
+
+      afterEach(function () {
+        $rootScope.$apply();
+      });
+
+      it('calls equivalent API method', function () {
+        promise.then(function () {
+          expect(LeaveRequestAPI.isValid).toHaveBeenCalled();
+        });
+      });
+
+      describe('when leave request is valid', function () {
+        it('returns no error', function () {
+          promise.then(function (result) {
+            expect(result).toEqual([]);
+          });
+        });
+
+        describe('when valid data not present', function () {
+          beforeEach(function () {
+            delete instance.contact_id;
+            promise = instance.isValid();
+          });
+
+          afterEach(function () {
+            $rootScope.$apply();
+          });
+
+          it('returns array of errors', function () {
+            promise.catch(function (result) {
+              expect(Object.keys(result).length).toBeGreaterThan(0);
+            });
+          });
+        });
+      });
+    });
+
+    describe('update()', function () {
+      var toAPIReturnValue = {
+          key: jasmine.any(String)
+        };
+
+      beforeEach(function () {
+        var defer = $q.defer();
```

Comment:
I think no need to create defer object, just calling `$rootScope.$apply();` would resolve the promise. Just like you did with `create` and `isValid`